app/javascript/components/map/MapStatistics.vue

```diff
@@ -319,7 +401,17 @@ export default {
         }
       })
     },
-    
+
+    waitUntilMapLoaded(callback, params, delay) {
```

Comment:
I see what you're trying to do with the recursive `setTimeout` loop but as you're not modifying the delay (as it's currently fixed at 100), I think it's better to use `setInterval` here unless you want to use a function to control the interval. It's also safer as it avoids a potential overflow